analytics.js

```diff
@@ -1185,15 +1192,24 @@ class Analytics {
 const instance = new Analytics();
 
 function processDataInAnalyticsArray(analytics) {
-  analytics.toBeProcessedArray.forEach((x) => {
-    const event = [...x];
-    const method = event[0];
-    event.shift();
-    // logger.debug("=====from analytics array, calling method:: ", method)
-    analytics[method](...event);
-  });
-
-  instance.toBeProcessedArray = [];
+  // Process the queue only if the default API is its first entry
+  // or SDK is already loaded
+  if (
+    analytics.toBeProcessedArray.length &&
+    (analytics.loaded || analytics.toBeProcessedArray[0][0] === defaultMethod)
+  ) {
```

Comment:
@tech-goldenphoenix , Here `defaultMethod` is `load`. No other event will be processed till `analytics.loaded` is `true`. So, the `analytics.toBeProcessedArray` will still have `event a` and `event b`.

Reply:
@MoumitaM Let's say the state appears that: 
```js
analytics.loaded = false
analytics.toBeProcessedArray = [['load'], ['track', 'event a'], ['track', 'event b']]
```
The `if` statement will be `true`. So the `while` statement will be executed so these following methods will be fired in synchronous order: 
`['load'] => ['track', 'event a'] => ['track', 'event b']`

But if the `load` method has to load the polyfill, the method `loadAfterPolyfill` will be called asynchronously. And the property `analytics.loaded` will be set to true after all the events fired
`['load'] => ['track', 'event a'] => ['track', 'event b'] => 'load' method load the polyfill asynchronously => analytics.loaded = true`

Based on the above flow, `event a` and `event b` will be ignored because the `analytics.loaded` is not set to true yet

Reply:
Apologies for the delay. Yes, you are right. Some events might be ignored depending on the polyfill and network strength. Refactored the code. 
Let us know if this works for you.